rfcs/52-unstable_features.md

```diff
@@ -0,0 +1,65 @@
+# Feature Name: `unstable_features`
+
+## Summary
+
+Accepting PRs for new features that may not have reach the high bar we set for Bevy, and disable their changes with a feature gate.
+
+## Motivation
+
+Some features would benefit from iterative work in Bevy, but that may be hard to do when a PRs need a high level of quality to be merged in.
+
+In this RFC, "feature" means the new feature or change under development, and "feature gate" the conditional compilation flag. "unstable" means a feature that is subject to change, not something that will break Bevy.
+
+## User-facing explanation
+
+So you want to contribute a new feature to Bevy! You have a few paths available to you, depending on the scope of the feature:
+- A small feature with a clear implementation? Go ahead and open that PR!
+- A large feature with an open design space that could use some discution about how to implement it? A RFC would be good.
+- Something in between that would benefit from iterative work? Here come the unstable feature workflow for you!
+
+You will need to open a tracking issue to explain the new feature you want to add. If it looks good, you'll be ready to open that PR with a feature gate for your changes!
+
+## Implementation strategy
+
+The new workflow for this kind of feature would be:
+- Submit an issue explaining the feature you want to add under a feature gate. This will be the tracking issue for stabilization once approved. This issue gets a new label `S-Stabilization`.
+- If approved, submit a PR with the new feature and the feature gate
+- This PR can be merged by someone with the merge rights on the related part of Bevy as soon as it's `S-Ready-For-Final-Review`, without an approval by the main Bevy maintainers.
+- Subsequent PRs on the feature must be linked to the tracking issue and follow the same process.
+- Once the feature is finished, enter the stabilization period. Open a PR that will remove the feature gate. This PR need to be approved by the main Bevy maintainers.
+
+An existing PR can be retroffited in this process on the suggestion of Bevy community members, and if the PR author agrees.
+
+The RFCs and the unstable features are closely related. An RFC can be implemented as an unstable feature to help resolve issues that need more experimentation. An unstable feature could require an RFC to reach stabilization.
+
+Each PR will still need to be approved by Bevy community members, and would still need to have the same level of quality as other PRs merged.
+
+The PRs for the initial feature, various changes during its finalization and for the stabilization can be opened by different persons.
+
+The feature gate should not be exposed on the main `bevy` crate, but only on the subcrates where it is relevant. It can be enabled as a user by adding dependencies directly on the subcrates and enabling the feature.
```

Comment:
How would we tackle breaking new ground with new official crates then? Should these be added as unstable features to the main bevy crate?

Reply:
No, a new crate can be added directly as an optional under feature gate dependency to the existing crate that need it, or directly as an added dependency. It shouldn't be any different than an existing subcrate

Reply:
Not sure if that's a good model to follow since many new crates will be building on top of existing foundational crates, not as dependencies for them. Or is the intention to have it added to bevy_internal in those cases?

Reply:
I think having as dependencies:
```
bevy = "0.6"
bevy_ecs = { version = "0.6", features = [ "unstable-4242-my-great-feature" ] }
bevy_new_crate = { version = "0.6" }
```

or targeting main
```
bevy = { git = "https://github.com/bevyengine/bevy" }
bevy_ecs = { git = "https://github.com/bevyengine/bevy", features = [ "unstable-4242-my-great-feature" ] }
bevy_new_crate = { git = "https://github.com/bevyengine/bevy" }
```

should work?

The other option would be indeed to raise all the unstable feature gates to `bevy_internal`, but I'm not sure that's worth the (very small) cost